tests/std/tests/P0896R4_ranges_alg_unique_copy/test.cpp

```diff
@@ -142,8 +143,21 @@ constexpr bool run_tests() {
     return true;
 }
 
+void test_gh1932() {
+    // Defend against regression of GH-1932, in which ranges::unique_copy instantiated
+    // iter_value_t<I> for a non-input iterator I.
+
+    std::istringstream str("42 42 42");
+    std::ostringstream result;
+    std::ranges::unique_copy(
+        std::istream_iterator<int>(str), std::istream_iterator<int>(), std::ostream_iterator<int>(result, " "));

```

Comment:
This test is already `using namespace std;`. Additionally, it would be nice to use braces to construct temporaries. Not worth resetting MSVC-PR testing, though.

Reply:
```suggestion
    istringstream str("42 42 42");
    ostringstream result;
    ranges::unique_copy(istream_iterator<int>{str}, istream_iterator<int>{}, ostream_iterator<int>{result, " "});
```